components/Main/index.js

```diff
@@ -1,22 +1,155 @@
+    const regex = /^[a-zA-Z0-9 ]*$/;
+    return regex.test(input);
+  }
+  let errorMSG = document.createElement("div");
+
+  errorMSG.style.color = "red";
+  errorMSG.style.fontSize = "11px";
+  errorMSG.style.fontWeight = "500";
+  errorMSG.style.position = "absolute";
+  errorMSG.style.left = " 305px";
+  errorMSG.style.top = " 80px";
 
-  proName.textContent = wantedProject.projectName;
+  const lableForProjectNameInput = document.querySelector(
+    ".lable-for-project-name-input"
+  );
+  EditProjectNameInput.onblur = function () {
+    if (!isValidString(EditProjectNameInput.value)) {
+      errorMSG.textContent = "*Please Enter a Valid Name Only 'a-z A-Z 0-9'";
+
+      EditProjectNameInput.style.border = "1px red solid";
+      EditProjectNameInput.before(errorMSG);
+    }
+
+    if (EditProjectNameInput.value == "") {
+      errorMSG.textContent = "*Project Name Can Not Be Empty'";
+
+      EditProjectNameInput.before(errorMSG);
+      EditProjectNameInput.style.border = "1px red solid";
+    }
+  };
+
+  EditProjectNameInput.onfocus = function () {
+    EditProjectNameInput.style.border = "";
+
+    errorMSG.remove();
+  };
+  //end of copied code
+
+  proName.replaceWith(EditProjectNameInput);
+  setTimeout(() => {
+    EditProjectNameInput.focus();
+  }, 0);
+  threeDotsIcon.classList.add("hide");
+  greenTrueIcon.classList.remove("hide");
+  redDeleteIcon.classList.remove("hide");
+
+  greenTrueIcon.addEventListener("pointerdown", () => {
+    if (!isValidString(EditProjectNameInput.value)) return;
+    let currentProjectId = getLocalStorage("currentProjectIdFromSidebar");
+
+    updateProjects(currentProjectId, EditProjectNameInput.value);
+    threeDotsIcon.classList.remove("hide");
+    greenTrueIcon.classList.add("hide");
+    redDeleteIcon.classList.add("hide");
+
+    proName.textContent = EditProjectNameInput.value;
+    previewProjectsNameOnTheSidebar();
+    const liElement = document.getElementById(currentProjectId);
+    liElement.classList.add("selected-project");
+
+    EditProjectNameInput.replaceWith(proName);
+  });
+  redDeleteIcon.addEventListener("pointerdown", () => {
+    threeDotsIcon.classList.remove("hide");
+    redDeleteIcon.classList.add("hide");
+    greenTrueIcon.classList.add("hide");
+    EditProjectNameInput.replaceWith(proName);
+  });
 };
 
-window.onload = () => {
-  let wantedProject =
-    getProjectFromLocalStorageById(
-      +getLocalStorage("currentProjectIdFromSidebar")
-    ) || "No Project";
+document.addEventListener("pointerdown", (e) => {
+  if (e.target.closest(".delete")) {
+    let currentId = getLocalStorage("currentProjectIdFromSidebar");
+
+    deleteProjectUiAction(currentId);
+  }
+  if (e.target.closest(".three-dots-icon")) {
+    actionMenuContainer.classList.toggle("show");
+  } else if (e.target.className != "three-dots-icon") {
+    actionMenuContainer.classList.remove("show");
+  }
+  if (e.target.closest(".edit")) {
+    editProjectUiAction();
+  }
+});
```

Comment:
1. Standardize event handling to avoid duplicate checks.
```js
    const { target } = event;
    const currentProjectId = getLocalStorage("currentProjectIdFromSidebar");

    if (target.closest(".delete")) {
        deleteProjectUIAction(currentProjectId);
    } else if (target.closest(".three-dots-icon")) {
        actionMenuContainer.classList.toggle("show");
    } else if (target.closest(".edit")) {
        editProjectUIAction();
    } else {
        actionMenuContainer.classList.remove("show");
    }
```

2. Use a single handleClick function to streamline conditions.
```js
document.addEventListener("pointerdown", handleClick);

function handleClick(event) {
```